src/panels/lovelace/editor/card-editor/hui-card-picker.ts

```diff
@@ -94,45 +110,83 @@ export class HuiCardPicker extends LitElement {
     }
 
     return html`
-      <search-input
-        .filter=${this._filter}
-        no-label-float
-        @value-changed=${this._handleSearchChange}
-        .label=${this.hass.localize(
-          "ui.panel.lovelace.editor.card.generic.search"
-        )}
-      ></search-input>
-      <div
-        id="content"
-        style=${styleMap({
-          width: this._width ? `${this._width}px` : "auto",
-          height: this._height ? `${this._height}px` : "auto",
-        })}
+      <mwc-tab-bar
+        .activeIndex=${this._tabIndex}
+        @MDCTabBar:activated=${(tabIndex) => {
+          this._tabIndex = tabIndex.detail.index;
+        }}
       >
```

Comment:
The tabs should be part of the header of the dialog, so they don't scroll with the content.